mongodb/mongodb_installer.sh

```diff
@@ -103,11 +120,87 @@ setup_mongodb()
         echo
     fi
 
-    echo "  -" >> $CONFIG_FILE
-    echo "    name: \"$LABEL\""               >> $CONFIG_FILE
-    echo "    url: \"$URL:$PORT\"" >> $CONFIG_FILE
-    echo "    user: \"$USER_NAME\""       >> $CONFIG_FILE
-    echo "    password: \"$PASSWORD\""        >> $CONFIG_FILE
+    if [ "$INITIAL_CHECK" == "initial" ]; then
+        ADMIN_BLOCK_NAME="  - name: \"$LABEL\""
+        ADMIN_BLOCK_HOST="    hostname: \"$URL\""
+        ADMIN_BLOCK_PORT="    port: $PORT"
+        ADMIN_BLOCK_USER="    username: \"$USER_NAME\""
+        ADMIN_BLOCK_PWD="    password: \"$PASSWORD\""
+        ADMIN_BLOCK_DB_NAME="    database: \"$DBNAME\""
+
+        ADMIN_DATA="$ADMIN_BLOCK_NAME\n$ADMIN_BLOCK_HOST\n$ADMIN_BLOCK_PORT\n$ADMIN_BLOCK_USER\n$ADMIN_BLOCK_PWD\n$ADMIN_BLOCK_DB_NAME"
+        ADMIN_DATA="$ADMIN_DATA\n  - ANOTHER-ADMIN-SERVER"
+        sed -i "0,/  - ANOTHER-ADMIN-SERVER/s//$ADMIN_DATA/" $CONFIG_FILE
+
+        DB_BLOCK_START="    databases:"
+        DB_BLOCK_NAME="    - name: \"$DBNAME\""
+        DB_BLOCK_USER="      username: \"$USER_NAME\""
+        DB_BLOCK_PWD="      password: \"$PASSWORD\""
+
+        DB_DATA="$ADMIN_BLOCK_NAME\n$ADMIN_BLOCK_HOST\n$ADMIN_BLOCK_PORT\n$DB_BLOCK_START"
+        DB_DATA="$DB_DATA\n    - ANOTHER-DB\n  - ANOTHER-DB-SERVER"
+        sed -i "0,/  - ANOTHER-DB-SERVER/s//$DB_DATA/" $CONFIG_FILE
+
+        DB_DATA="$DB_BLOCK_NAME\n$DB_BLOCK_USER\n$DB_BLOCK_PWD"
+        DB_DATA="$DB_DATA\n    - ANOTHER-DB"
+        sed -i "0,/    - ANOTHER-DB/s//$DB_DATA/" $CONFIG_FILE
+    else
+        DB_DATA="$DB_BLOCK_NAME\n$DB_BLOCK_USER\n$DB_BLOCK_PWD"
+        DB_DATA="$DB_DATA\n    - ANOTHER-DB"
+        sed -i "0,/    - ANOTHER-DB/s//$DB_DATA/" $CONFIG_FILE
+    fi
+}
+
+setup_mongodb()
+{
+    sed -i "s/    - ANOTHER-DB//g" $CONFIG_FILE
```

Comment:
Before setting a new mongodb server we clear the placeholder for adding "ANOTHER-DB" because a DB can not be setup without setting "ANOTHER-SERVER" FIRST